runtime/vam/op/over.go

```diff
@@ -0,0 +1,163 @@
+package op
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/runtime/vam/expr"
+	"github.com/brimdata/zed/vector"
+)
+
+type Over struct {
+	zctx   *zed.Context
+	parent vector.Puller
+	exprs  []expr.Evaluator
+
+	vecs []vector.Any
+	idx  uint32
+}
+
+func NewOver(zctx *zed.Context, parent vector.Puller, exprs []expr.Evaluator) *Over {
+	return &Over{
+		zctx:   zctx,
+		parent: parent,
+		exprs:  exprs,
+	}
+}
+
+func (o *Over) Pull(done bool) (vector.Any, error) {
+	if done {
+		o.vecs = nil
+		return o.parent.Pull(true)
+	}
+	for {
+		if len(o.vecs) == 0 || o.idx >= o.vecs[0].Len() {
+			vec, err := o.parent.Pull(done)
+			if vec == nil || err != nil {
+				return nil, err
+			}
+			o.vecs = o.vecs[:0]
+			for _, e := range o.exprs {
+				vec2 := e.Eval(vec)
+				vec2 = vector.Apply(true, func(vecs ...vector.Any) vector.Any { return vecs[0] }, vec2)
```

Comment:
What is the point of the apply here?

Reply:
`e.Eval(vec)` can return a vector.Dynamic containing vector.Unions, and over needs to "deunion" them. Is there any other way than this?

For completeness, here's the failure that appears without this.

```console
$ go test . -run /runtime/ztests/op/
--- FAIL: TestZed (0.23s)
    --- FAIL: TestZed/runtime/ztests/op (0.01s)
        --- FAIL: TestZed/runtime/ztests/op/over-union (0.01s)
            ztest.go:409: runtime/ztests/op/over-union.yaml: === vector ===
                expected and actual output differ:
                --- expected
                +++ actual
                @@ -1,6 +1,5 @@
                 1
                 "foo"
                -3
                -4
                -"bar"
                +[3,4]((ip,[int64]))
                +"bar"((int64,string))

FAIL
FAIL	github.com/brimdata/zed	0.829s
FAIL
```

Reply:
Makes sense